### ISSUE TITLE
Не принимает код маркировки пива.

### ISSUE THREAD

**Issue by kkmserver** [shtrih-m/fr_drv_ng#282]
Вот такой КМ: 0104680046264732\u1d215l>VEUc\u1d93Is54
Дает ошибку: 211-ККТ: Код маркировки не распознан 
Этот же КМ на Атолах, Дримкасс - проходит без проблем.
И КМ других видов товаров проходят.
ШТРИХ-ON-LINE, Firmware: 62597 от 18.01.2024, FFD: 1.2

Что можно сделать?

**Comment by surfernsk** [comment]
@kkmserver Проверьте корректность этого Кода маркировки. В этом вам поможет документация на сайте честного знака. И в некоторых ситуациях поможет приложение от Честного знака на телефоне.

P.S.

КМ предоставленный Вами не корректен! 
Ваш КМ:
![image](https://github.com/shtrih-m/fr_drv_ng/assets/2264893/c968207f-ded1-456d-a0e0-d2e09e6f0b2d)

И как должно быть:
![image](https://github.com/shtrih-m/fr_drv_ng/assets/2264893/95478e95-f9d5-4fbe-987d-cbc30cfede35)

**Comment by surfernsk** [comment]
**А если владельцу ККТ все равно что отправлять в ОФД , Честный знак, РосПотребНадзор и др**. -  тогда можно включить игнорирование этой ошибки. Чтобы игнорировать ошибку установите настройку Т1П57=1 "ПРИНИМАТЬ ВСЕ КОДЫ ТОВАРА".

**Comment by gainup** [comment]
> @kkmserver Проверьте корректность этого Кода маркировки. В этом вам поможет документация на сайте честного знака. И в некоторых ситуациях поможет приложение от Честного знака на телефоне.
> 
> P.S.
> 
> КМ предоставленный Вами не корректен! Ваш КМ: ![image](https://private-user-images.githubusercontent.com/2264893/317814386-c968207f-ded1-456d-a0e0-d2e09e6f0b2d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTU0MjYsIm5iZiI6MTcxMTY1NTEyNiwicGF0aCI6Ii8yMjY0ODkzLzMxNzgxNDM4Ni1jOTY4MjA3Zi1kZWQxLTQ1NmQtYTBlMC1kMmUwOWU2ZjBiMmQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjhUMTk0NTI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTViMWRmN2FjZDRlOWI1ZmY3OWQ4OTM1MjRkZTk5OGEzZjVkZTZkMTJlZWZmMWQyZjQ0MjZiNTZlYzE5ODcxNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ._16rqljPKtki8oncqQgZtKoT-QDKbGU4ZW-FDlGyOnU)
> 
> И как должно быть: ![image](https://private-user-images.githubusercontent.com/2264893/317813927-95478e95-f9d5-4fbe-987d-cbc30cfede35.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTE2NTU0MjYsIm5iZiI6MTcxMTY1NTEyNiwicGF0aCI6Ii8yMjY0ODkzLzMxNzgxMzkyNy05NTQ3OGU5NS1mOWQ1LTRmYmUtOTg3ZC1jYmMzMGNmZWRlMzUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMjhUMTk0NTI2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZWU2YjUwMTVkODFiOGI3Zjg3YjgyMDU5Mzk1MjNkOWFjNjVlNTFhMTZkNGEyZWY0ZGEzODdjZjRkZGIxNmI1OSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.TSNrjvguciL723bcvUeWCLaT7YJLfo7-1_xrRCqd2iI)

Да, вы правы. Там лишний разделитель. Но 211 ошибка и на корректных кодах маркировки (как у вас ниже) с частичным выбытием (пиво в кегах). Проблема только в Штрих-Онлайн, на других моделях те же коды проходят без проблем.

**Comment by kkmserver** [comment]
Сори, я не ту строку вставил. Вот это не проходит:
0104680046264732215l>VEUc\u001d93Is54
Или так 0104680046264732215l>VEUc\<GS>93Is54
Именно так как и у Вас.
Но ККТ не принимает его. дает эту 211 ошибку...

**Comment by olefard** [comment]
Нужен лог драйвера.

**Comment by kkmserver** [comment]
И пиво не бутылочное а весовое...

**Comment by kkmserver** [comment]
Работаем не через драйвер.
Вот что шлется в порт по каманде FF-61:
<     02-39-FF-61-1E-00-00-00-01-00-20-0F-30-31-30-34-36-38-30-30-34-36-32-36-34-37-33-32-32-31-35-6C-3E-56-45-55-63-1D-39-33-49-73-35-34-3C-08-01-00-29-FF-03-01-00-31-0D-05-01-00-31-1D
Ответ:  06-02-03-FF-61-D3-4E

**Comment by olefard** [comment]
В TLV что-то странное:
3C-08-01-00-29 - тег 2108=литры - норм
FF-03-01-00-31-0D-05-01-00-31 - а что здесь? тег 1023 сформирован неправильно. И тег 1293 здесь лишний.

Если хотите продать 1 литр, то первый байт(положение десятичной точки) = 0, второй байт = 1
То есть будет как-то так:
3C 08 01 00 29 FF 03 02 00 00 01

**Comment by kkmserver** [comment]
1023 - FVLN - все сразу стало на место, СПС!!!

**Comment by kkmserver** [comment]
Теперь почему-то принимает этот КМ разливного пива только когда количество = 1.
При допустим продаже 2 л.: 
- команда FF61 с тегом 1023 = 3.0 проходит.
- а при регистрации чека в команде FF67 дает ошибку "(94) ККТ: Допустимо только единичное количество"

Если тот-же чек делать через ДТО - то та-же самая ошибка.
Не подскажите направление куда копать?

**Comment by olefard** [comment]
В поле планируемый статус нужно передавать 2(мерный товар, в стадии реализации).
У вас было 1(штучный товар, реализован).

**Comment by kkmserver** [comment]
СПС,
Да я и передавал 2108 = 2.
Оказывается имеет значение порядок передачи этих тегов.
А так как тегов много - довольно не очевидно в каком порядке их передавать.
СПС!!!

**Comment by olefard** [comment]
Не-не, в команде поле после пароля - планируемый статус, надо в нем 2 передать

Такая посылка должны быть:
FF-61-1E-00-00-00-**02**-00-20-0B-30-31-30-34-36-38-30-30-34-36-32-36-34-37-33-32-32-31-35-6C-3E-56-45-55-63-1D-39-33-49-73-35-34-3C-08-01-00-29-FF-03-02-00-00 03-КС

**Comment by kkmserver** [comment]
В FF-61 я передавал двойку.
А вот в чеке сначала шла команда FF67 и только потом 2108.
Поменял местами - все заработало.

**Comment by kkmserver** [comment]
Теперь другое - КМ маркировки разливного пива проходят только если в чеке одна строка с таким КM
Иначе выдает 211-ККТ: Код маркировки не распознан
В автоле проходит указание одного и того-же КМ в нескольких строках.
В штрихах это можно как-то сделать?

**Comment by swex** [comment]
> Теперь другое - КМ маркировки разливного пива проходят только если в чеке одна строка с таким КM Иначе выдает 211-ККТ: Код маркировки не распознан В автоле проходит указание одного и того-же КМ в нескольких строках. В штрихах это можно как-то сделать?

В прошивках после 18.01.2024 теперь так: 
```
Разрешено один и тот же КМ выбывать несколько раз в одном чеке при любых настройках, если мера количества не штуки.
```
PS: в этом репо всё таки поддержка драйвера, по общим вопросам лучше пишите на [tp@shtrih-m.ru](mailto:tp@shtrih-m.ru).
